src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -186,61 +186,75 @@
              [:div {:style {:margin "0.5rem 0 0 1rem"}}
               (this :render-details (:method entity))
               [:div {:style {:fontWeight 500 :marginTop "1rem"}} "WDL:"]
               [CodeMirror {:text (get-in entity [:method :payload])}]]
              [CodeMirror {:text (:payload entity)}]))])])
    :render-details
-   (fn [{:keys [props refs]} entity]
-     (let [{:keys [editing?]} props
+   (fn [{:keys [props refs state]} entity]
+     (let [{:keys [editing? redacted?]} props
+           redacted-snapshot (:redacted-snapshot @state)
            make-field
            (fn [key label & {:keys [dropdown? wrap? render]}]
              [:div {:style {:display "flex" :alignItems "baseline" :paddingBottom "0.25rem"}}
               [:div {:style {:flex "0 0 100px" :fontWeight 500}} (str label ":")]
               [:div {:style {:flex "1 1 auto" :overflow "hidden" :textOverflow "ellipsis"
                              :whiteSpace (when-not wrap? "nowrap")}}
                (if (and editing? dropdown?)
-                 (style/create-identity-select {:ref key
-                                                :data-test-id (config/when-debug "edit-method-config-snapshot-id-select")
-                                                :style {:width 100}
-                                                :defaultValue (key entity)
-                                                :onChange (when-let [f (:onSnapshotIdChange props)]
-                                                            #(f (int (common/get-text refs "snapshotId"))))}
-                                               (:snapshots props))
+                 (style/create-identity-select-name {:ref key
+                                                     :data-test-id (config/when-debug "edit-method-config-snapshot-id-select")
+                                                     :style {:width 120}
+                                                     :defaultValue (if redacted-snapshot -1 (key entity))
+                                                     :onChange (when-let [f (:onSnapshotIdChange props)]
+                                                                 #(f (int (common/get-text refs "snapshotId"))))}
+                                                    (:snapshots props)
+                                                    redacted-snapshot)
                  (let [rendered ((or render identity) (key entity))]
                    [:span {:title rendered} rendered]))]])]
        [:div {}
         [:div {:style {:display "flex"}}
          [:div {:style {:flex "1 1 50%" :paddingRight "0.5rem"}}
+          (when redacted?
+            [:div {:style {:fontWeight 500 :paddingBottom "0.25rem"}}
+             (icons/icon {:style {:color (:warning-state style/colors)}} :warning) " Snapshot Redacted"])
           (make-field :namespace "Namespace")
           (make-field :name "Name")
           (make-field :snapshotId "Snapshot ID" :dropdown? true)
```

Comment:
I think this should be `{:dropdown? true}` on the end here, not your code, but still

Reply:
that didn't work. i'm guessing the param list interprets the rest of the args as a map - still weird.